### ISSUE TITLE
context sensitive  ->New  in the project view is very slow

### ISSUE THREAD

**Issue by dv99156** [JustSid/AfterglowIntelliJ#5]
It takes 4-6  seconds for the menu render.  Moving the focus through the menu items off New is also slow.  Tested on a java maven project  idea 15.0.3.  The menu is instantaneous without the plugin installed. 

![image](https://cloud.githubusercontent.com/assets/5826364/12480386/fdb2b1fa-c00f-11e5-98ed-3f350ea62f16.png)

**Comment by JustSid** [comment]
Interesting, there seem to be some broken icons in there as well.. I'll see what I can do, thanks for the report!

**Comment by Lewenhaupt** [comment]
I'm also having this problem. Didn't have it about two weeks ago but now I do, and unfortunately I had to unistnall the plugin because it got on my nerves too much.

**Comment by JustSid** [comment]
@Lewenhaupt and @davidvol, I've implemented a caching mechanism and have tested it with a larger CLion project on Windows which I was experiencing slowdowns with previously. I only have Windows 10 though which previously showed different performance characteristics than say Windows 8(.1), so if you guys would test it as well I would really appreciate it. I've uploaded a pre-compiled jar here: http://cl.widerwille.com/fWmK

Sorry for the long delay in actually getting behind a fix here, I'm a OS X and iOS developer and it was working fine on OS X, and I have also no idea how to profile Java so I was basically just stabbing in the dark and gave up pretty quickly.
